tools/windows-node-installer/test/e2e/azure_test.go

```diff
@@ -107,20 +112,43 @@ var (
 	credentials *types.Credentials
 )
 
+// getAzureCloudProvider returns an azure cloud provider
+func getAzureCloudProvider() (cloudprovider.Cloud, error) {
+	cloud, err := cloudprovider.CloudProviderFactory(kubeconfig, azureCredentials, "", artifactDir, azureImageID,
```

Comment:
We should not use `CloudProviderFactory` method here, as we are using service account to create oc client for CI. We need to do it similar to [aws_test](https://github.com/openshift/windows-machine-config-bootstrapper/blob/master/tools/windows-node-installer/test/e2e/aws_test.go#L111)

Reply:
I agree that we should be using the service account to create the oc client for CI.

However the actual function that needs to be tested is `cloudprovider.CloudProviderFactory` as that is what we use in the operator. Directly calling the azure cloud provider is skipping testing a portion of the code.

My intention here was to change as little as possible with the end to end tests to make them actually work. Previously we were not using the service account to create an oc client, I feel that adding that is outside the scope of this ticket.